NitroxServer/Communication/Packets/PacketHandler.cs

```diff
@@ -67,27 +35,35 @@ public void Process(Packet packet, Connection connection)
         
         private void ProcessAuthenticated(Packet packet, Player player)
         {
-            PacketProcessor packetProcessor;
-            if (authenticatedPacketProcessorsByType.TryGetValue(packet.GetType(), out packetProcessor))
+            try
             {
-                packetProcessor.ProcessPacket(packet, player);
+                Type serverPacketProcessorType = typeof(AuthenticatedPacketProcessor<>);
+                Type packetType = packet.GetType();
+                Type packetProcessorType = serverPacketProcessorType.MakeGenericType(packetType);
+
+                PacketProcessor processor = (PacketProcessor)NitroxServiceLocator.LocateService(packetProcessorType);
+                processor.ProcessPacket(packet, player);
             }
-            else
+            catch
             {
-                defaultPacketProcessor.ProcessPacket(packet, player);
+                new DefaultServerPacketProcessor(NitroxServiceLocator.LocateService<PlayerManager>()).ProcessPacket(packet, player);
```

Comment:
Since PlayerManager is needed in multiple places in this class it's better to pass it as a parameter through the constructor of `PacketHandler` class and initialize this class with `NitroxServiceLocator`.